packages/jest-config/src/index.js

```diff
@@ -19,12 +19,17 @@ import normalize from './normalize';
 function readConfig(
   argv: Argv,
   packageRoot: string,
+  // Whether it needs to look into `--config` arg passed to CLI.
+  // It only used to read initial config. If the initial config contains
+  // `project` property, we don't want to read `--config` value and rather
+  // read individual configs for every project.
+  dontUseArgvConfig?: boolean,
```

Comment:
i really want to rewrite this file and the part of `jest-cli` that resolves/reads config. This function should not be used for every config resolution (default config resolution, `--config file.js`, `--config '{"json": true}', and MPR configs) because they have different logic and using the same flow for everything will cause a lot of bugs.
this little hack should work for now though